### PR TITLE
migrate modification from master into 2.19.03/lts

### PR DESCRIPTION
**Pull request by Lordess**
apply commits following 2.19.03/lts tag of master.

### CODE REVIEW

**Comment by yanhuqing666** [comment]
This branch has conflicts that must be resolved。。。

File: 1.config_file/1.6_cache/1.6.1_cache_setting.md [hunk 1]
```diff
@@ -3,9 +3,10 @@
 
  dble的cache使用有如下三类：
 
-*   SQLRouteCache：缓存sql语句到数据所在节点集的映射。
-*   ER_SQL2PARENTID：缓存子表分区发现sql语句到数据所在节点的映射。
-*   TableID2DataNodeCache：缓存分区键值到数据所在节点的映射。这是一个层次cache。
+*   SQLRouteCache：从前端连接收到的SQL以及对应的路由结果 内容： shema_user_SQL -> 具体路由结果RouteResult
+*   ER_SQL2PARENTID：父子表辅助查询SQL以及对应的路由结果。在插入ER子表的时候需要根据子表joinKey(父表parentKey)计算它应该插入的结点，所以需要辅助路由来查询，然后将辅助路由及对应的结果缓存下来以备下次查询。内容为：schema:select * from 父表 where parentKey = (value of joinKey) -> 对应数据dataNode
+*   TableID2DataNodeCache：表格主键(schema.xml中声明的主键)和对应数据节点  内容 tableName -> primaryKeyValue -> 对应数据dataNode
+
```
Comment (Lordess) [inline 1.config_file/1.6_cache/1.6.1_cache_setting.md:9]: This conflict is safe. It comes from b45377936a76f0eca851de443388c8ec9b823fdf

File: 2.Function/2.10_table_meta/2.10.4_View_Meta.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 #### 2.10.4.1 view meta概述
 在Dble 2.18.11.0 版本中新添加了对于view的支持，作为一个中间件支持view采用的方法是和复杂查询类似的逻辑，将VIEW创建的语句中的select部分进行解析，并将解析所得到的解析树进行存储，当有query调用到视图的时候使用解析树进行局部的替代还原成完整的查询SQL，从而达到view实现的效果。  
 
-在Mysql中的View是一种持久化的存储，结合在Dble中的view实现方式，Dble中view的实现会给Dble本身带来状态，这个状态需要Dble自身进行保存和维护，所以在view中采取了和XA事务日志相似的方式，并且以视图创建SQL的形式进行以下两种方式的存储  
+在MySQL中的View meta信息是一种持久化的存储。Dble中的view实现方式也类似，这样Dble中view的实现会给Dble本身带来状态，这个状态需要Dble自身进行保存和维护，所以在view中采取了和XA事务日志相似的方式，并且以视图创建SQL的形式进行以下两种方式的存储
```
Comment (Lordess) [inline 2.Function/2.10_table_meta/2.10.4_View_Meta.md:6]: This conflict is safe. It comes from e3c925aef7003238b45186d38b0ebe744898462e

File: 2.Function/2.20_slow_query_log.md [hunk 3]
```diff
@@ -746,7 +746,12 @@ select * from sharding_two_node where id =1\G
 
 ```
 
+#### 2.20.3.3 限制
+
+当mysqldumpslow版本为5.6或更早的版本时，在解析慢日志的过程中，由于mysqldumpslow只能识别'190428 10:28:16'格式的Time字段，而dble却使用了'2019-04-28T10:28:16.515000Z'字段格式，因此解析生成的慢日志会出错。详情参见issue:<https://github.com/actiontech/dble/issues/908>
+
```
Comment (Lordess) [inline 2.Function/2.20_slow_query_log.md:752]: This conflict is safe. It comes from cce94eb3f50b23f131601ae74cfa42a17fda8010

File: 3.SQL_Syntax/3.2_DML/3.2.09_LOAD_DATA.md [hunk 4]
```diff
@@ -28,9 +28,16 @@ INFILE 'file_name' INTO TABLE tbl_name
...
 
-#### 3.2.9.3 限制
+#### 3.2.9.3 原理
+
+dble解析MySQL协议之后，会根据数据路由拆分文件,每满足maxRowSizeToFile(可通过server.xml配置)就写到文件中，再通过 load data local infile 的方式导入到后端结点。    
+所以，这里[local_infile](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_local_infile)这个参数会影响到load data的正确性.  
+可参考[#1085](https://github.com/actiontech/dble/issues/1085)   
+
+
+#### 3.2.9.4 限制
```
Comment (Lordess) [inline 3.SQL_Syntax/3.2_DML/3.2.09_LOAD_DATA.md:38]: This conflict is safe. It comes from bca4cc369fcfa565872b1af9887f9db8d253dd8a

File: 9.Sysbench_samples/9.2_conf.md [hunk 5]
```diff
@@ -67,7 +67,7 @@
     <user name="test">
         <property name="password">111111</property>
         <property name="schemas">mytest</property>
-        <property name="benchmark">1000000</property>
+        <property name="maxCon">1000000</property>
```
Comment (Lordess) [inline 9.Sysbench_samples/9.2_conf.md:70]: This conflict is safe. It comes from 7e4880bc7729bf125a1da92f4f41ed1a7cabb396

**Review by Lordess — COMMENTED** [review]
Reviewed all conflicting files. They are safe. Ask for merging.

**Comment by yanhuqing666** [comment]
It's not a safety problem. 
Conflicts MUST be resolved, or I can't merge either. It's Github's rule

**Review by yanhuqing666 — APPROVED** [review]
ok

**Comment by Lordess** [comment]
Resolved conflicts :-)

**Pull request merged**